### PR TITLE
Added EDA notebook

### PR DESCRIPTION
**Pull request by soumik12345**

### CODE REVIEW

**Review by sayakpaul — CHANGES_REQUESTED** [review]
Looks good to me with a few comments:

* Add sufficient annotation above the non-trivial code cells. 
* [This notebook](https://www.kaggle.com/kmader/point-cloud-overview) does provide a flexible template to perform basic EDA on point cloud segmentation. We may want to follow that. 
* There seems to be a mismatch in the mapping:

image

**Comment by soumik12345** [comment]
@sayakpaul fixed the mismatch in mapping, will continue with the EDA.

**Comment by sayakpaul** [comment]
@soumik12345 things are looking better. Also, try adding the legends in the overlaid plot to make it easier to debug and visualize better.

**Comment by soumik12345** [comment]
@sayakpaul , although the 1st point cloud labels are being plotted accurately, the later ones seem to be wrong. Can you please suggest what could be the issue?

File: pointnet_shapenet.ipynb [hunk 1]
```diff
@@ -0,0 +1,383 @@
+{
```
Comment (sayakpaul) [inline pointnet_shapenet.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.        point_cloud = np.loadtxt(points_files[0])
</pre><p> </p><p>Why are we loading the same thing again and again inside the loop?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/point-cloud-segmentation/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (sayakpaul) [inline pointnet_shapenet.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.        labels = os.listdir(labels_dir)
</pre><p> </p><p>Same. If I am reading it correctly, it's the same thing being loaded iteratively. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/point-cloud-segmentation/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (sayakpaul) [inline pointnet_shapenet.ipynb:1]: <p>I would also investigate a couple more to see if this inconsistency is largely the case. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/point-cloud-segmentation/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (soumik12345) [reply pointnet_shapenet.ipynb:1]: @sayakpaul this was the culprit, fixed it

Comment (soumik12345) [reply pointnet_shapenet.ipynb:1]: Remove this redundancy

Comment (soumik12345) [reply pointnet_shapenet.ipynb:1]: @sayakpaul it was a silly mistake on my part fixed it, there's no inconsistancy.

**Review by sayakpaul — CHANGES_REQUESTED** [review]
@soumik12345 much better now. 

Here are a few minor suggestions:

* If possible, then add more details (like which labels did you include, homepage link of the ShapeNet dataset, etc.) in the description of the [released dataset](https://github.com/soumik12345/point-cloud-segmentation/releases/tag/v0.1). 
* Add legends in the visualizations. 
* To make it easier for us and also the other users, consider grouping together similar blocks of code and add descriptive comments wherever necessary. Referring to the code cell where data is being prepared. 
* After the data arrays have been prepared consider taking individual elements from them and comment about their shapes, how they are represented, and so on. 

A related question:

I suppose for modeling, `point_clouds` and `point_cloud_labels` are necessary and the rest are for visualization purposes?

**Comment by soumik12345** [comment]
> I suppose for modeling, point_clouds, and point_cloud_labels are necessary and the rest are for visualization purposes?

@sayakpaul you are correct.

**Comment by soumik12345** [comment]
> If possible, then add more details (like which labels did you include, homepage link of the ShapeNet dataset, etc.) in the description of the released dataset.

@sayakpaul updated the release description.

**Review by sayakpaul — CHANGES_REQUESTED** [review]
@soumik12345 I still don't see any useful comments or investigation of the arrays formed after their formation. Without that, it's hard for me to understand how you are approaching the preprocessing steps.

The visualizations look amazing with `plotly`.

**Comment by soumik12345** [comment]
> After the data arrays have been prepared consider taking individual elements from them and comment about their shapes, how they are represented, and so on.

Hi @sayakpaul , commented about the arrays being generated.

**Review by sayakpaul — APPROVED** [review]
@soumik12345 thank you! 

Since we are now using `test_point_clouds`, therefore, to avoid confusion, I suggest we name the other arrays accordingly. `train_ point_clouds`, for example.

**Comment by soumik12345** [comment]
> @soumik12345 thank you!
> 
> Since we are now using `test_point_clouds`, therefore, to avoid confusion, I suggest we name the other arrays accordingly. `train_ point_clouds`, for example.

Hi @sayakpaul , I added a validation split at the end.

**Comment by soumik12345** [comment]
@sayakpaul  Let's merge this branch and create a new branch for the dataloader.

**Pull request merged**